Include/Rover/Evaluator.hpp

```diff
@@ -43,24 +62,25 @@ namespace Rover {
   template<typename T>
   template<typename U>
   Evaluator::Entry<T>::Entry(U&& value)
-      : m_value(std::forward<U>(value)) {}
+    : m_value(std::forward<U>(value)) {}
 
   template<typename Generator>
   Evaluator::Evaluation::Evaluation(Generator& generator,
       std::unique_ptr<BaseEntry> entry)
-      : m_generator(&generator),
-        m_entry(std::move(entry)) {}
+    : m_generator(&generator),
+      m_entry(std::move(entry)) {}
 
   template<typename Generator>
   typename Generator::Type Evaluator::evaluate(Generator& generator) {
     using Type = typename Generator::Type;
+    auto type_erasing_ptr = Details::TypeErasingPtr(&generator);
     auto i = std::find_if(m_evaluations.begin(), m_evaluations.end(),
-      [&] (const auto& evaluation) {
-        return evaluation.m_generator == &generator;
+      [&type_erasing_ptr](const auto& e) {
```

Comment:
Use a capture all by ```&```. Only use specific captures when you need to mix some captures by reference and others by value.